src/test/java/io/github/externschool/planner/service/RoleServiceImplTest.java

```diff
@@ -0,0 +1,52 @@
+package io.github.externschool.planner.service;
+
+import io.github.externschool.planner.entity.Role;
+import io.github.externschool.planner.exceptions.RoleNotFoundException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
```

Comment:
[Can be used when a test focuses **only** on JPA components.](https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/test/autoconfigure/orm/jpa/DataJpaTest.html). JPA means Java Persistence API, it's used for data sources AKA repositories, not for services.  
@SpringBootTest is better choice.